isotope-intensity-estimation.Rmd

```diff
@@ -284,23 +307,28 @@ frmls2 <- apply(counts_df, 1, function(x) paste0(sapply(seq_along(x),
 ```
 ---->
 
-```{r, eval = !file.exists(paste0("data/RData/isotope-intensity-estimation/"))}
+```{r, eval = !file.exists(paste0(dr, "cnf_frequencies.RData"))}
 data(isotopes)
-chemforms <- cmps$formula
+## Ensure that element order is correct
+chemforms <- sapply(cmps$formula, standardizeFormula)
 threshold = 0.01
 iso_ps <- isopattern(isotopes, chemforms , threshold = threshold)
 # iso_ps <- isopattern(isotopes, c("Fe", "C4H3BrN2") , threshold = threshold)
 
+## Process the isotope pattern to define a character string for each
+## isotopologue - ensuring the order of isotopes to be the same across
+## compounds
+isos <- c("13C", "2H", "15N", "17O", "18O", "33S", "34S", "35S", "36S", 
+          "36Cl", "37Cl")
 cnf_per_cmpd <- lapply(iso_ps, function(iso_p) {
-  tmp <- iso_p[, intersect(colnames(iso_p), c("13C", "2H", "15N", "17O", "18O", 
-                                            "33S", "34S", "35S", "36S", 
-                                            "36Cl", "37Cl")), drop = FALSE]
-  apply(tmp, 1, 
-        function(row) paste(sapply(names(row)[which(row > 0)], 
-                                   function(nm) paste0(row[nm],"[", nm, "]")),
-                            collapse = ""))
+    idx <- match(colnames(iso_p), isos)
```

Comment:
I think way we avoid the problem that e.g. Cl comes before N in one formula and after N in another - but please have also a look to cross-check/validate if that's true @andreavicini